src/Pact/Eval.hs

```diff
@@ -282,45 +282,53 @@ evaluateConstraints
   :: Info
   -> Module
   -> HM.HashMap Text Ref
-  -> Eval e ()
+  -> Eval e (Module, HM.HashMap Text Ref)
 evaluateConstraints info Interface{} _ =
   evalError info $ "Unexpected: interface found in module position while solving constraints"
-evaluateConstraints info Module{..} evalMap = foldMap evaluateConstraint _mInterfaces
+evaluateConstraints info m evalMap =
+  -- we would like the lazy semantics of foldr to shortcircuit the solver
+  foldr evaluateConstraint (pure (m, evalMap)) (_mInterfaces m)
   where
-    evaluateConstraint ifn = do
-      irefs <- preview $ eeRefStore . rsModules . ix ifn . mdRefMap
-      case irefs of
+    evaluateConstraint ifn em = do
+      (Module n ks meta c h b ifs, refMap) <- em
```

Comment:
So this is kind of janky because we don't have prisms for `Module`. That would be the solution to this, however.

Reply:
I don't think I understand the problem or the solution. What's wrong with field accessors? (`mod{_mMeta=meta <> _interfaceMeta}`)

Reply:
You can do that? oh man. I should do that.

Reply:
I've actually just gone ahead with a lensy version that cleaned things up quite a bit